SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java

```diff
@@ -1381,12 +1383,29 @@ public void event(String name) {
     @Override
     public void event(String name, Map<String, String> payload) {
         try {
-            _event(name, payload);
+            if(isValidEventName(name)) {
+                _event(name, payload);
+            }
         } catch (Exception e) {
             SwrveLogger.e(LOG_TAG, "Exception thrown in Swrve SDK", e);
         }
     }
 
+    private boolean isValidEventName(String name) {
+        List<String> restrictedNamesStartWith = new ArrayList<String>() {{
+            add("Swrve.");
+            add("swrve.");
+        }};
+
```

Comment:
Not a requirement but this is a handy one I use for list initialization: ` List<String> restrictedNamesStartWith = Arrays.asList("Swrve.", "swrve.");`